### PR TITLE
Added unit test with 304 response

### PR DESCRIPTION
**Pull request by Haimantika**
Solves https://github.com/microsoft/kiota-http-go/issues/163

### CODE REVIEW

**Review by andrueastman — COMMENTED** [review]
Any chance you can double check the build issue @Haimantika

**Comment by Haimantika** [comment]
> Any chance you can double check the build issue @Haimantika

The new commit should solve it.

File: nethttp_request_adapter.go [hunk 1]
```diff
@@ -743,7 +743,7 @@ const ErrorBodyFoundAttributeName = "com.microsoft.kiota.error.body_found"
 func (a *NetHttpRequestAdapter) throwIfFailedResponse(ctx context.Context, response *nethttp.Response, errorMappings abs.ErrorMappings, spanForAttributes trace.Span) error {
 	ctx, span := otel.GetTracerProvider().Tracer(a.observabilityOptions.GetTracerInstrumentationName()).Start(ctx, "throwIfFailedResponse")
 	defer span.End()
-	if response.StatusCode < 400 {
+	if response.StatusCode < 400 || response.StatusCode == nethttp.StatusNotModified {
```
Comment (calebkiage) [inline nethttp_request_adapter.go:746]: Isn't 304 included in the range of values <400? I don't see why explicitly checking for 304 would change the behavior here?

Comment (Haimantika) [reply nethttp_request_adapter.go:746]: Ideally, yes. So should we skip it?

Comment (andrueastman) [reply nethttp_request_adapter.go:746]: @Haimantika I think what we should is probably revert this then add a unit test that captures the scenario that is decribed in https://github.com/microsoft/kiota-http-go/issues/163 in this file

https://github.com/microsoft/kiota-http-go/blob/e626a135a64edc797b104ed8d767053abda04ec4/nethttp_request_adapter_test.go#L50

Comment (Haimantika) [reply nethttp_request_adapter.go:746]: This is helpful! Thanks @andrueastman I will work on it.

**Comment by baywet** [comment]
@Haimantika gentle reminder. Are you still working on this?

**Pull request closed without merging**